test/functional/unified-spec-runner/match.ts

```diff
@@ -235,32 +248,70 @@ export function specialCheck(
   }
 }
 
+// CMAP events where the payload does not matter.
+const EMPTY_CMAP_EVENTS = {
+  poolCreatedEvent: ConnectionPoolCreatedEvent,
+  poolClosedEvent: ConnectionPoolClosedEvent,
+  connectionCreatedEvent: ConnectionCreatedEvent,
+  connectionReadyEvent: ConnectionReadyEvent,
+  connectionCheckOutStartedEvent: ConnectionCheckOutStartedEvent,
+  connectionCheckOutFailedEvent: ConnectionCheckOutFailedEvent,
+  connectionCheckedOutEvent: ConnectionCheckedOutEvent,
+  connectionCheckedInEvent: ConnectionCheckedInEvent
+};
+
+function validEmptyCmapEvent(
+  expected: ExpectedCommandEvent | ExpectedCmapEvent,
+  actual: CommandEvent | CmapEvent
+) {
+  return Object.keys(EMPTY_CMAP_EVENTS).some(key => {
+    const eventType = EMPTY_CMAP_EVENTS[key];
+    return actual instanceof eventType;
+  });
+}
+
 export function matchesEvents(
-  expected: ExpectedEvent[],
-  actual: CommandEvent[],
+  expected: (ExpectedCommandEvent & ExpectedCmapEvent)[],
+  actual: (CommandEvent & CmapEvent)[],
   entities: EntitiesMap
 ): void {
-  // TODO: NodeJS Driver has extra events
-  // expect(actual).to.have.lengthOf(expected.length);
+  if (actual.length !== expected.length) {
+    // const actualNames = actual.map(a => a.constructor.name);
+    // const expectedNames = expected.map(e => Object.keys(e)[0]);
+    // TODO: NodeJS Driver has extra events
+    // expect(actual).to.have.lengthOf(expected.length, `Expected event count mismatch, expected ${inspect(expectedNames)} but got ${inspect(actualNames)}`);
+  }
 
   for (const [index, actualEvent] of actual.entries()) {
     const expectedEvent = expected[index];
 
-    if (expectedEvent.commandStartedEvent && actualEvent instanceof CommandStartedEvent) {
+    if (expectedEvent.commandStartedEvent) {
+      expect(actualEvent).to.be.instanceOf(CommandStartedEvent);
       resultCheck(actualEvent, expectedEvent.commandStartedEvent, entities, [
         `events[${index}].commandStartedEvent`
       ]);
-    } else if (
-      expectedEvent.commandSucceededEvent &&
-      actualEvent instanceof CommandSucceededEvent
-    ) {
+    } else if (expectedEvent.commandSucceededEvent) {
+      expect(actualEvent).to.be.instanceOf(CommandSucceededEvent);
       resultCheck(actualEvent, expectedEvent.commandSucceededEvent, entities, [
         `events[${index}].commandSucceededEvent`
       ]);
-    } else if (expectedEvent.commandFailedEvent && actualEvent instanceof CommandFailedEvent) {
+    } else if (expectedEvent.commandFailedEvent) {
+      expect(actualEvent).to.be.instanceOf(CommandFailedEvent);
       expect(actualEvent.commandName).to.equal(expectedEvent.commandFailedEvent.commandName);
+    } else if (validEmptyCmapEvent(expectedEvent, actualEvent)) {
```

Comment:
I would suggest moving this into L313 instead, as the final clause `else if (!validEmptyCmapEvent...) { expect.fail ... }`, that would avoid having an odd empty block in the middle of otherwise uniform checks

Reply:
what do you think is the benefit of having it as a separate block as:
```
    } else if (validEmptyCmapEvent(expectedEvent, actualEvent)) {
      // This should just always pass since the event must exist and match the type.
    } else {
      expect.fail(`Events must be one of the known types, got ${inspect(actualEvent)}`);
    }
```
vs just
```
    } else if (!validEmptyCmapEvent(expectedEvent, actualEvent)) {
      expect.fail(`Events must be one of the known types, got ${inspect(actualEvent)}`);
    }
    // remove final else
```

Reply:
Updated - nice catch.